terraform/gcp/cluster.tf

```diff
@@ -70,10 +70,42 @@ resource "google_container_cluster" "cluster" {
 
   cluster_autoscaling {
     # This disables node autoprovisioning, not cluster autoscaling!
-    enabled = false
+    enabled = var.enable_node_autoprovisioning
     # Use a scheduler + autoscaling profile optimized for batch workloads like ours
     # https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-autoscaler#autoscaling_profiles
     autoscaling_profile = "OPTIMIZE_UTILIZATION"
+
+    // When node auto-provisioning is enabled, set the min and max amount of
+    // CPU and memory to allocate
+    dynamic "resource_limits" {
+      for_each = var.enable_node_autoprovisioning ? [1] : []
+
+      content {
+        resource_type = "memory"
+        minimum       = var.min_memory
+        maximum       = var.max_memory
+      }
+    }
+
+    dynamic "resource_limits" {
+      for_each = var.enable_node_autoprovisioning ? [1] : []
+
+      content {
+        resource_type = "cpu"
+        minimum       = var.min_cpu
+        maximum       = var.max_cpu
+      }
+    }
+
+    // When node auto-provisioning is enabled, define the service account to be
+    // used by the node VMs
+    dynamic "auto_provisioning_defaults" {
+      for_each = var.enable_node_autoprovisioning ? [1] : []
```

Comment:
Is this explicitly required by terraform? If so, let's make a note here. Otherwise, is the default different from what we set here?

Reply:
It's not required. I set it to mimic what the Pangeo Makefile is doing (specifically using the `--autoprovisioning-service-account` flag. It's unclear to me what the default will be, maybe just using the `pangeo-hubs-cluster-sa` account?

Reply:
I will remove it for now and deploy - we can always add it back in later.